src/Compilers/CSharp/Portable/Binder/DecisionDagBuilder.cs

```diff
@@ -459,6 +459,14 @@ private static void Assert(bool condition, string message = null)
                         MakeTestsAndBindings(output, pattern, tests, bindings);
                     }
                 }
+                else if (Binder.IsZeroElementTupleType(inputType))
+                {
+                    // Work around https://github.com/dotnet/roslyn/issues/20648: The compiler's internal APIs such as `declType.IsTupleType`
```

Comment:
>// Work around https://github.com/dotnet/roslyn/issues/20648: [](start = 20, length = 61)

nit: Consider removing this comment since it is already captured in `Binder.IsZeroElementTupleType`